anxcloud/data_source_dns_zones.go

```diff
@@ -18,14 +20,23 @@ func datasourceDNSZones() *schema.Resource {
 }
 
 func dataSourceDNSZonesRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	c := m.(providerContext).legacyClient
-	a := zone.NewAPI(c)
+	a := m.(providerContext).api
 
-	zones, err := a.List(ctx)
+	var oc types.ObjectChannel
+	err := a.List(ctx, &clouddnsv1.Zone{}, api.ObjectChannel(&oc))
```

Comment:
not part of the review, just making sure

You know you can not only `List()` with an `ObjectChannel` but also `Paged`? ^^